### PR TITLE
Add `ibm-entitlement-key` as default image pull secret

### PR DESCRIPTION
**Pull request by amdabhad**
- Code changes for part 1 of https://github.com/IBM/ibm-spectrum-scale-csi/issues/632
	 * Add ibm-entitlement-key as another default image pull secret
	 * Keep default pull secrets even when user specifies the pull secrets
	 * Removed duplicate setting of pull secrets 

- Part 2 (check for pull failures and include a message to the user for debug) is WIP - The state of pods changes from ImagePullBackOff to CrashLoopBackOff to ImagePullErr - so the pull error is logged only sometimes currently - will create a separate PR for this.

## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

-

## How risky is this change?
- [ ] Small, isolated change
- [ ] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

File: operator/controllers/syncer/csi_node.go [hunk 1]
```diff
@@ -103,8 +103,9 @@ func (s *csiNodeSyncer) SyncCSIDaemonsetFn(daemonSetRestartedKey string, daemonS
 			secrets = append(secrets, corev1.LocalObjectReference{Name: s})
 		}
 	} else {
-		// Use default imagePullSecret
-		secrets = append(secrets, corev1.LocalObjectReference{Name: config.DefaultImagePullSecret})
+		// Use default imagePullSecrets
+		secrets = append(secrets, corev1.LocalObjectReference{Name: config.ImagePullSecretRegistryKey},
+			corev1.LocalObjectReference{Name: config.ImagePullSecretEntitlementKey})
```
Comment (Jainbrt) [inline operator/controllers/syncer/csi_node.go:108]: I think we need ibm-spectrum-scale-csi-registrykey,ibm-entitlement-key in any case even if user has defined the pull secret in the cr object. @deeghuge any input ?

**Review by Jainbrt — CHANGES_REQUESTED** [review]

File: operator/controllers/syncer/csi_node.go (hunk 1 above)
Comment (amdabhad) [reply operator/controllers/syncer/csi_node.go:108]: As discussed earlier, the current implementation of imagePullSecrets is such that: if user specifies imagePullSecrets from driver CR, we are not using default imagePullSecrets. Let me post a message on our slack channel to discuss if we want to change this behavior and close on this quickly.

Comment (deeghuge) [reply operator/controllers/syncer/csi_node.go:108]: yes, both secret must be there always so else part can be removed

Comment (deeghuge) [reply operator/controllers/syncer/csi_node.go:108]: secret always will be ImagePullSecretRegistryKey, ImagePullSecretEntitlementKey, <user defined secrets>

Comment (amdabhad) [reply operator/controllers/syncer/csi_node.go]: Changed this behavior

**Review by Jainbrt — APPROVED** [review]

**Review by deeghuge — APPROVED** [review]

**Pull request merged**